src/yunohost/settings.py

```diff
@@ -0,0 +1,189 @@
+import os
+import json
+import errno
+
+from datetime import datetime
+from collections import OrderedDict
+
+from moulinette.core import MoulinetteError
+from moulinette.utils.log import getActionLogger
+
+logger = getActionLogger('yunohost.settings')
+
+SETTINGS_PATH = "/etc/yunohost/settings.json"
+SETTINGS_PATH_OTHER_LOCATION = "/etc/yunohost/settings-%s.json"
+
+# a settings entry is in the form of:
+# name: {type, value, default, description, [possibilities]}
+# possibilities is only for enum
+
+# description is implied from the translated strings
+# the key is "global_settings_setting_%s" % key.replace(".", "_")
+
+# type can be:
+# * bool
+# * int
+# * string
+# * enum (in form a python list)
+
+# we don't store the value in default options
+DEFAULTS = OrderedDict([
```

Comment:
We want to enforce a fixed list of settings, putting it directly into the code is a good way to do that.

Furthermore, it allows to use OrderedDirect which is impossible in json (and super annoying to do in yaml and yaml is slow).

I also have some problems with the "one day we might have" which leads to over engineered code for situations where a later modification is trivial, so if it's a problem one day, we'll fix it this day :)